tests/helpers/perf_run_helper.rb

```diff
@@ -944,75 +945,253 @@ def get_process_hash(perf_result_processes)
-           "GROUP BY pe_build_number, test_scenario)"
-
-      data = query_bigquery sql
-      if !data.empty?
-        logger.info("Baseline result returned from BigQuery: #{data}")
-      else
-        logger.error("Cannot find result that matches query: #{sql}")
-      end
-      data[0]
+      logger.error("Cannot find result that matches query: #{sql}")
+      result = {}
     end
+    return result
   end
 
   def baseline_assert(atop_result, gatling_result)
-    process_results = get_process_hash(atop_result.processes)
-
-    # Handle 3 different things: avg_response_time which comes from gatling result,
-    # global atop results and per process atop results
     baseline = get_baseline_result
-    baseline&.each do |key, value|
-      # orchestration has a jump in memory usage in 2019.2 due to plan executor
-      # https://tickets.puppetlabs.com/browse/BOLT-986
-      # Currently we don't have a way to whitelist that... so just turning off the check
-      # Ticket to make a way https://tickets.puppetlabs.com/browse/SLV-515
-      next if key.to_s == "process_orchestration_services_release_avg_mem" && BEAKER_PE_VER =~ /^2019.2*/
-
-      assert_value = if key.to_s == "avg_response_time"
-                       gatling_result.avg_response_time.to_f
-                     elsif key.to_s.start_with? "process"
-                       process_results[key.to_s].to_f
-                     else
-                       atop_result.send(key).to_f
-                     end
-
-      puts "Value for #{key} is baseline: #{value} Actual: #{assert_value}"
-      next if key.to_s.start_with?("process") && key.to_s.end_with?("cpu")
-
-      max_baseline_variance = \
-        if key.to_s == "process_orchestration_services_release_avg_mem"
-          15
-        else
-          10
-        end
-      next unless value.is_a? Integer
+    delta_data = baseline_to_results_delta(baseline, atop_result, gatling_result)
+    assert_later(validate_baseline_delta(delta_data), "Things are really fucked up now")
+  end
 
-      assert_later((assert_value - value) / value * 100 <= max_baseline_variance,
-                   "The value of #{key} '#{assert_value}' " \
-                   "was not within #{max_baseline_variance}% " \
-                   "of the baseline '#{value}'")
+  # rubocop: enable Naming/AccessorMethodName
+
+  # Calculate the variances on the data set provided and return a hash of the
+  # failing elements with the variance calculation appended to each key.
+  #
+  # @param  [Hash] data  Data to validate { key: [v0, v2], ... }
+  #
+  # @return [Hash]  Hash of failing performance keys associated with there values and variance
+  def find_failing_variances(data)
+    deltas = data.transform_values { |v| v[1].to_f / v[0] }
+    failures = deltas.select { |_k, v| (1 - v).abs > MAX_BASELINE_VARIANCE }
+    # exception for PROC_ORCH_REL_MEM
+    if failures.include? PROC_ORCH_REL_MEM
```

Comment:
also, since we are refactoring this code.  It would be better if the override was like 5% (and could be negative) and it meant add/subtract 5% to the baseline then compare.  The difference is that a result that is 15% below the baseline should get flagged...

Reply:
I will disagree when it comes to defining this value as an adder to the normal variance (i.e. 5% rather than 15%).  To me this would be more confusing and would muddy the waters if we chose to define something with a stricter than normal variance.

All of the variance comparisons occur in either direction.  Any variance either more or less will fail the variance logic.

I don't quite understand what you have in mind for the dynamic looping for `MAX_BASELINE_VARIANCE_PROC_ORCH_REL_MEM`.  Can you provide more detail to your thoughts on this?

Reply:
what you say is true.  Really there are 2 types of overrides than.  One to change the overall variance from the default. This is really a configuration type setting, so not really an override.  The other is to shift the whole range based on a root change to the perf.  What we did for orch was the first one, increase the variance range... because it was the easiest thing we could do at the time.  But it is technically wrong.  Say a metrics was 10, and we accepted plus or minus 1 (10%).  9 would be okay.  But they added something new in the latest release, so that 12 was the nominal value.  What we did to accommodate that was change the plus or minus to 3... allowing for 13 (10+3)... but now also allowing for 7 (10-3).  Which we shouldn't have done.  If seven was worth flagging before it should still be worth flagging now.  The correct way would be to apply a +2 to the base and keep the variance the same.  Your refactor makes this not so hard to do... and I was suggesting we do it... or accommodate doing it with some minor changes...

Reply:
as for dynamic, which figures into my above novel...  when calculating the delta, you could add in a base adjustment if a variable like BASE_ADJUSTMENT_<key>  where key is like PROC_ORCH_REL_MEM which I think is also the key to the data hash that is passed in.

Reply:
dynamic part 2... Currently you do the variance check against max variance.  Then check the <key> against a hardcoded key value, and if found redo the variance check with a hardcoded variable override.  I am saying, name the variable VARIANCE_ADJUSTMENT_<key>.  Then before doing the first variance check, look to see if a VARIANCE_ADJUSTMENT_<key> exists for the key you are considering.  And if so, use the value of that VARIANCE_ADJUSTMENT_<key> variable instead of MAX_BASELINE_VARIANCE.  By doing this... all we have to do to add a new variance override is define a new variable.  As it is right now, we have to add a new variable, and a hardcoded conditional to figure out when to recalculate and possibly delete the failure...

Reply:
created https://tickets.puppetlabs.com/browse/SLV-699 for the baseline adjust feature.

Reply:
Using the defined constants to dynamically assign variance maximum thresholds has been implemented and added to this PR.